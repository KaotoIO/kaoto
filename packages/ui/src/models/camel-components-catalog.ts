import { CatalogKind } from './catalog-kind';

export interface ICamelComponentDefinition {
  component: ICamelComponent;
  componentProperties: ICamelComponentProperties;
  headers: ICamelComponentHeaders;
  properties: ICamelComponentProperties;
}

export interface ICamelComponent {
  kind: CatalogKind.Component;
  name: string;
  title: string;
  description: string;
  deprecated: boolean;
  firstVersion?: string;
  label: string;
  javaType?: string;
  supportLevel?: string;
  groupId?: string;
  artifactId?: string;
  version: string;
  scheme?: string;
  extendsScheme?: string;
  syntax?: string;
  async?: boolean;
  api?: boolean;
  consumerOnly?: boolean;
  producerOnly?: boolean;
  lenientProperties?: boolean;
}

export interface CatalogItemCommon {
  index: number;
  kind: string;
  displayName: string;
  group: string;
  label: string;
  required: boolean;
  javaType: string;
  deprecated: boolean;
  deprecationNote: string;
  autowired: boolean;
  secret: boolean;
  description: string;
}

export interface ICamelComponentProperties {
  type: string;
}

export interface ICamelComponentHeaders extends CatalogItemCommon {
  constantName: string;
}
