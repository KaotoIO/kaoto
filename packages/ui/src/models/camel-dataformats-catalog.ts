import { ICamelProcessorModel, ICamelProcessorProperty } from './camel-processors-catalog';
import { KaotoSchemaDefinition } from './kaoto-schema';

export interface ICamelDataformatDefinition {
  model: ICamelDataformatModel;
  properties: Record<string, ICamelDataformatProperty>;
  propertiesSchema: KaotoSchemaDefinition['schema'];
}

export type ICamelDataformatModel = ICamelProcessorModel;

export type ICamelDataformatProperty = ICamelProcessorProperty;
