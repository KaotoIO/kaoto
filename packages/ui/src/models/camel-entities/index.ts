export * from './base-entity';

/** Camel Route definitions */
export * from './flows/route';
