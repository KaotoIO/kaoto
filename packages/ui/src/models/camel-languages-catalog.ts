import { JSONSchemaType } from 'ajv';
import { ICamelProcessorModel, ICamelProcessorProperty } from './camel-processors-catalog';

export interface ICamelLanguageDefinition {
  model: ICamelLanguageModel;
  properties: Record<string, ICamelLanguageProperty>;
  propertiesSchema: JSONSchemaType<unknown>;
}

export interface ICamelLanguageModel extends ICamelProcessorModel {}

export interface ICamelLanguageProperty extends ICamelProcessorProperty {}
