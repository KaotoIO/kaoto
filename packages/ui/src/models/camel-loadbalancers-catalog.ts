import { JSONSchemaType } from 'ajv';
import { ICamelProcessorModel, ICamelProcessorProperty } from './camel-processors-catalog';

export interface ICamelLoadBalancerDefinition {
  model: ICamelLoadBalancerModel;
  properties: Record<string, ICamelLoadBalancerProperty>;
  propertiesSchema: JSONSchemaType<unknown>;
}

export interface ICamelLoadBalancerModel extends ICamelProcessorModel {}

export interface ICamelLoadBalancerProperty extends ICamelProcessorProperty {}
