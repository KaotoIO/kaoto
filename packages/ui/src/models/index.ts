export * from './camel-catalog-index';
export * from './camel-components-catalog';
export * from './camel-processors-catalog';
export * from './catalog-kind';
export * from './flow';
export * from './kamelets-catalog';
export * from './schema';
