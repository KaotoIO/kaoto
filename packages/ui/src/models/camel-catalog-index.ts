import { ICamelComponentDefinition } from './camel-components-catalog';
import { ICamelProcessorDefinition } from './camel-processors-catalog';
import { CatalogKind } from './catalog-kind';
import { IKameletDefinition } from './kamelets-catalog';

export interface CamelCatalogIndex {
  catalogs: Catalogs;
  schemas: CatalogEntry[];
}

export interface Catalogs {
  models: CatalogEntry;
  components: CatalogEntry;
  languages: CatalogEntry;
  dataformats: CatalogEntry;
  kamelets: CatalogEntry;
}

export interface CatalogEntry {
  name: string;
  version: string;
  file: string;
}

export type CatalogTypes = Record<string, ICamelComponentDefinition | ICamelProcessorDefinition | IKameletDefinition>;

export interface ComponentsCatalog {
  [CatalogKind.Component]?: Record<string, ICamelComponentDefinition>;
  [CatalogKind.Processor]?: Record<string, ICamelProcessorDefinition>;
  [CatalogKind.Kamelet]?: Record<string, IKameletDefinition>;
  [key: string]: unknown;
}

export interface CatalogCamelComponent {
  type: CatalogKind.Component;
  definition: ICamelComponentDefinition;
}

export interface CatalogCamelProcessor {
  type: CatalogKind.Processor;
  definition: ICamelProcessorDefinition;
}

export interface CatalogKamelet {
  type: CatalogKind.Kamelet;
  definition: IKameletDefinition;
}
