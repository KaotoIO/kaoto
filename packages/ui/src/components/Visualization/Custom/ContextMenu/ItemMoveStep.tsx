import { ContextMenuItem } from '@patternfly/react-topology';
import { FunctionComponent, PropsWithChildren } from 'react';
import { IDataTestID } from '../../../../models';
import { AddStepMode, IVisualizationNode } from '../../../../models/visualization/base-visual-entity';
import { useMoveStep } from '../hooks/move-step.hook';

interface ItemMoveStepProps extends PropsWithChildren<IDataTestID> {
  mode: AddStepMode.AppendStep | AddStepMode.PrependStep;
  vizNode: IVisualizationNode;
}

export const ItemMoveStep: FunctionComponent<ItemMoveStepProps> = (props) => {
  const { canBeMoved, onMoveStep } = useMoveStep(props.vizNode, props.mode);

  if (!canBeMoved) return null;

  return (
    <ContextMenuItem onClick={onMoveStep} data-testid={props['data-testid']}>
      {props.children}
    </ContextMenuItem>
  );
};
