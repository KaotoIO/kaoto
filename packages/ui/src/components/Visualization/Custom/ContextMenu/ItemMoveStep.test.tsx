import { AngleDoubleDownIcon, AngleDoubleUpIcon } from '@patternfly/react-icons';
import { fireEvent, render } from '@testing-library/react';
import { AddStepMode, createVisualizationNode } from '../../../../models';
import { useMoveStep } from '../hooks/move-step.hook';
import { ItemMoveStep } from './ItemMoveStep';

// Mock the `useMoveStep` hook
jest.mock('../hooks/move-step.hook', () => ({
  useMoveStep: jest.fn(),
}));

describe('ItemMoveStep', () => {
  const vizNode = createVisualizationNode('test', {});
  const mockOnMoveStep = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render Move Next ContextMenuItem', () => {
    // Mock the `useMoveStep` hook to return compatible state
    (useMoveStep as jest.Mock).mockReturnValue({
      onMoveStep: mockOnMoveStep,
      canBeMoved: true,
    });

    const { container } = render(
      <ItemMoveStep vizNode={vizNode} mode={AddStepMode.AppendStep}>
        <AngleDoubleDownIcon /> Move Next
      </ItemMoveStep>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should not render Move Before ContextMenuItem', () => {
    // Mock the `useMoveStep` hook to return compatible state
    (useMoveStep as jest.Mock).mockReturnValue({
      onMoveStep: mockOnMoveStep,
      canBeMoved: false,
    });

    const { container } = render(
      <ItemMoveStep vizNode={vizNode} mode={AddStepMode.AppendStep}>
        <AngleDoubleUpIcon /> Move Before
      </ItemMoveStep>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should call onMoveStep when the context menu item is clicked', () => {
    // Mock the `useMoveStep` hook to return compatible state
    (useMoveStep as jest.Mock).mockReturnValue({
      onMoveStep: mockOnMoveStep,
      canBeMoved: true,
    });

    const wrapper = render(
      <ItemMoveStep vizNode={vizNode} mode={AddStepMode.AppendStep}>
        <AngleDoubleDownIcon /> Move Next
      </ItemMoveStep>,
    );
    fireEvent.click(wrapper.getByText('Move Next'));

    expect(mockOnMoveStep).toHaveBeenCalledTimes(1);
  });
});
