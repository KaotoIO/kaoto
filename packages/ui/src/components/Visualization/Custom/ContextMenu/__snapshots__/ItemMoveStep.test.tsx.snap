// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemMoveStep should not render Move Before ContextMenuItem 1`] = `<div />`;

exports[`ItemMoveStep should render Move Next ContextMenuItem 1`] = `
<div>
  <li
    class="pf-v6-c-menu__list-item"
    data-ouia-component-id="OUIA-Generated-DropdownItem-1"
    data-ouia-component-type="PF6/DropdownItem"
    data-ouia-safe="true"
    role="none"
  >
    <button
      class="pf-v6-c-menu__item"
      role="menuitem"
      tabindex="-1"
      type="button"
    >
      <span
        class="pf-v6-c-menu__item-main"
      >
        <span
          class="pf-v6-c-menu__item-text"
        >
          <svg
            aria-hidden="true"
            class="pf-v6-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 320 512"
            width="1em"
          >
            <path
              d="M143 256.3L7 120.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0L313 86.3c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.4 9.5-24.6 9.5-34 .1zm34 192l136-136c9.4-9.4 9.4-24.6 0-33.9l-22.6-22.6c-9.4-9.4-24.6-9.4-33.9 0L160 352.1l-96.4-96.4c-9.4-9.4-24.6-9.4-33.9 0L7 278.3c-9.4 9.4-9.4 24.6 0 33.9l136 136c9.4 9.5 24.6 9.5 34 .1z"
            />
          </svg>
           Move Next
        </span>
      </span>
    </button>
  </li>
</div>
`;
